### ISSUE TITLE
Install failing on Windows due to fsevents

### ISSUE THREAD

**Issue by williamluke4** [adrianmcli/truffle-next#21]
```javascript
λ truffle unbox adrianmcli/truffle-next
Downloading...
Unpacking...
Setting up...
Error: Command failed: cd client && npm install && npm run link-contracts
npm WARN client@1.0.0 No repository field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.3 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

ln: creating symbolic link `contracts' to `../../build/contracts': No such file or directory
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! client@1.0.0 link-contracts: `cd lib && ln -s ../../build/contracts contracts`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the client@1.0.0 link-contracts script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Work\AppData\Roaming\npm-cache\_logs\2018-01-14T15_52_15_906Z-debug.log

    at ChildProcess.exithandler (child_process.js:272:12)
    at ChildProcess.emit (events.js:160:13)
    at maybeClose (internal/child_process.js:943:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:220:5)
```

**Comment by adrianmcli** [comment]
@williamluke4 thanks for reporting the issue. I haven't used Windows in almost 10 years, so I'm not sure how much I can help here. But it does look like the issue lies in the creation of the symlink. Unfortunately, I have no idea how symlinks would work with Windows.

**Comment by adrianmcli** [comment]
I should mention that you can just manually create the symlink and everything should work just fine.

**Comment by adrianmcli** [comment]
If anyone with a Windows computer is interested in helping, please submit a PR.

**Comment by lightswarm124** [comment]
Don't have a windows computer but perhaps this could provide some answers

https://stackoverflow.com/a/5549583

**Comment by adrianmcli** [comment]
@lightswarm124 thanks, I think we'll need someone with an actual Windows machine to test out a `cross-env` way of doing this.

**Comment by adrianmcli** [comment]
We can probably use this package: https://github.com/charlesguse/run-script-os

We'll use a `mklink` command if the user's computer is a Windows machine, and for everything else, we will use the same `ln -s` command. This is a pretty straight-forward thing to implement, we just need someone with a Windows machine.

**Comment by williamluke4** [comment]
#29 Fix Submitted but needs testing on Mac and Linux

**Comment by adrianmcli** [comment]
Unfortunately the fix from #29 removes the symlink and thus makes it not work on Mac and Linux. #29 also conflates this issue with trying to add Typescript to the project. As a result, I have closed #29.
